### PR TITLE
Added "interactable" attribute to portals which can disable the player portal interaction

### PR DESCRIPTION
**Pull request by benfah**
Hi,
I added this feature especially for the "look-through" portals, which imo shouldn't feature the custom manipulation through the portal, so I added a boolean field which can prevent this and implemented it in the mirror class. No other portal is affected :)

### CODE REVIEW

**Comment by LoganDark** [comment]
Mirrors are meant to be reach-through. Please implement this as an NBT option only and I will consider merging it

File: src/main/java/com/qouteall/immersive_portals/portal/Mirror.java [hunk 1]
```diff
@@ -15,6 +15,7 @@ public Mirror(EntityType<?> entityType_1, World world_1) {
     public void tick() {
         super.tick();
         teleportable = false;
+        interactable = false;
```
Comment (LoganDark) [inline src/main/java/com/qouteall/immersive_portals/portal/Mirror.java]: Delete this line, mirrors are meant to be reach-through

**Review by LoganDark — APPROVED** [review]
Looks good to me. Needs @qouteall approval

**Comment by benfah** [comment]
Found that kind of confusing that the mirror was meant to be reach-through tbh, is this just a personal preference or something else?

**Comment by LoganDark** [comment]
> is this just a personal preference or something else?

Of both myself and @qouteall. He implemented subroutines into the code that specifically detect mirrors and alter the raycasting slightly so you can target blocks through them.

**Comment by benfah** [comment]
Oh, ok.

**Comment by LoganDark** [comment]
Maybe it could be a config option to not aim through mirrors? This can definitely be altered client-side. You could open another PR for it, or just edit this one. You'd just add some code to `canInteractThroughPortal` that checks the config option and you're good to go.

**Comment by benfah** [comment]
Sure. Should I open another PR or add that to the current PR here?

**Comment by LoganDark** [comment]
You can add it here. It should be somewhat straightforward.

**Comment by benfah** [comment]
Is this fine?

**Comment by LoganDark** [comment]
Yep, looks good to me. Let me make a couple more review comments on your other code.

File: src/main/java/com/qouteall/immersive_portals/portal/Portal.java [hunk 2]
```diff
@@ -49,7 +49,9 @@
     public Quaternion rotation;
     
     public double motionAffinity = 0;
-    
+
+    public boolean interactable = true;
```
Comment (LoganDark) [inline src/main/java/com/qouteall/immersive_portals/portal/Portal.java]: I would recommend making this private and adding a setter method, to discourage developers (especially third party ones using this mod as an API) from accessing it directly.

File: src/main/java/com/qouteall/immersive_portals/portal/Portal.java [hunk 3]
```diff
@@ -165,7 +171,11 @@ public boolean isCullable() {
     public boolean isTeleportable() {
         return teleportable;
     }
-    
+
+    public boolean canInteractThroughPortal() {
```
Comment (LoganDark) [inline src/main/java/com/qouteall/immersive_portals/portal/Portal.java]: `getInteractable`, and you should add `setInteractable`

A short javadoc (`/**` comment) would be helpful too.

**Review by LoganDark — CHANGES_REQUESTED** [review]
Only gripes I have with this PR. High quality code, considering what you have to integrate with 😅

File: src/main/java/com/qouteall/immersive_portals/portal/Portal.java (hunk 3 above)
Comment (LoganDark) [reply src/main/java/com/qouteall/immersive_portals/portal/Portal.java]: Additionally, you actually forgot to respect the config option you just added!

Perhaps add an override under the mirror class?

Comment (LoganDark) [reply src/main/java/com/qouteall/immersive_portals/portal/Portal.java]: whoops, missed the thing under tick(), but perhaps an override would be more appropriate. Return `interactable` if people have the config option turned on, but if they have it turned off, return false. That way you don't have to mutate the data every tick

**Review by LoganDark — APPROVED** [review]
Looks good to me :D

**Pull request merged**